tock/hours/tests/test_views.py

```diff
@@ -367,6 +368,69 @@ def test_prefilled_timecard(self):
         )
         self.assertEqual(prefilled_hours, {None})
 
+    def test_add_hours_with_existing_tco(self):
+        """
+        Test that addHours increases the current timecard by the
+        specified amount
+        """
+        self.user = self.regular_user
+
+        new_period = hours.models.ReportingPeriod.objects.create(
+            start_date=datetime.date(2016, 1, 1),
+            end_date=datetime.date(2016, 1, 7),
+        )
+        new_timecard = hours.models.Timecard.objects.create(
+            user=self.user,
+            submitted=False,
+            reporting_period=new_period
+        )
+        new_project = projects.models.Project.objects.get(name="Midas")
+        new_tco = hours.models.TimecardObject.objects.create(
+            timecard=new_timecard,
+            project=new_project,
+            hours_spent=10.12
+        )
+
+        url = "%s?project=%d&hours=2" % (reverse('AddHours'), new_project.id)
+
+        response = self.app.get(
+            url, None,
+            headers={'X_AUTH_USER': self.user.email}
+        )
+
+        self.assertEqual(response.status_code, 302)
+        new_tco.refresh_from_db()
+        self.assertEqual(new_tco.hours_spent, Decimal('12.12'))
+
+    def test_add_hours_without_tco(self):
+        """
+        Test that addHours creates a new timecardobject when
+        none is available
+        """
+        self.user = self.regular_user
+
+        new_project = projects.models.Project.objects.get(name="Midas")
+        new_period = hours.models.ReportingPeriod.objects.create(
+            start_date=datetime.date(2016, 1, 1),
+            end_date=datetime.date(2016, 1, 7),
+        )
+        new_timecard = hours.models.Timecard.objects.create(
+            user=self.user,
+            submitted=False,
+            reporting_period=new_period
+        )
+
+        url = "%s?project=%d&hours=2" % (reverse('AddHours'), new_project.id)
+
+        old_count = new_timecard.timecardobjects.count()
+        response = self.app.get(
+            url, None,
+            headers={'X_AUTH_USER': self.user.email}
+        )
+        new_count = new_timecard.timecardobjects.count()
+        self.assertEqual(old_count + 1, new_count)
```

Comment:
In addition, testing the redirect response info and to see if the correct message is included.